pkg/controller/nodes/task/handler.go

```diff
@@ -561,6 +561,24 @@ func (t Handler) Handle(ctx context.Context, nCtx handler.NodeExecutionContext)
 	// So now we will derive this from the plugin phase
 	// TODO @kumare re-evaluate this decision
 
+	var inputs *core.LiteralMap
+	if ts.PluginPhase == pluginCore.PhaseUndefined && t.eventConfig.RawOutputPolicy == controllerConfig.RawOutputPolicyInline {
+		// The task should only reach undefined exactly once. Since we want to send the inputs inline at some point in the task execution flow (but not necessarily every event), we send them for this event transition only.
+		// The calls to read from the catalog below may call InputReader.Get subsequent times, but the underlying implementation uses a CachedInputReader that will
+		// not re-download the inputs for the duration of this Handle call.
+		tk, err := tCtx.tr.Read(ctx)
+		if err != nil {
+			logger.Errorf(ctx, "failed to read TaskTemplate, error :%s", err.Error())
+			return handler.UnknownTransition, err
+		}
+		if tk.Interface != nil {
```

Comment:
Check that len(Interface.Inputs.Variables) > 0 ?

Reply:
done